### PR TITLE
Issue 14: Docker image prepare (#14)

### LINKED ISSUE #14: setup docker file
##Problem description
This repository produces bindings which are used in container deployments using docker images. This issue tracks the setup changes required to put the binding jars in the images  so that they can be deployed as enhanced Pravega images.

they should be making a new Pravega image with only additonal jars, little or no additional files if possible

##Problem location
the docker file and associated build script

##Suggestions for an improvement

until this repository goes public, we can use internal image repositories but it should switch to Pravega and Travis will need the credentials to push the images there.

### PR DESCRIPTION
**Pull request by ravibeta**
**Change log description**  
Prepares the binding jars in the images so that they can be deployed on containers in K8s or Docker.
Fixes #14

**Purpose of the change**  
Fixes #14

**What the code does**  
It prepares and pushes the docker image so that they can be included as additional storage adapters. 

**How to verify it**  
- point the registry/repository to private
- run ./gradlew docker and ./gradlew dockerPush

### CODE REVIEW

**Review by sachin-j-joshi — APPROVED** [review]
Approving to unblock testing

File: gradle/maven-publish.gradle [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+                    developer {
+                        id = 'ravibeta'
+                        name = 'Ravi Rajamani'
+                    }
+                }
+            }
+
+            // Declares coordinates for publishing
+            groupId = 'io.pravega'
+            artifactId = "pravega-bindings-" + project.name.replace(':', '-')
```
Comment (ravibeta) [inline gradle/maven-publish.gradle]: no rename

**Review by sachin-j-joshi — COMMENTED** [review]
1. Use the built jar as is without renaming it.
2. From the original image, remove original bindings jar.

**Review by sachin-j-joshi — CHANGES_REQUESTED** [review]
Use the built jar as is without renaming it.
From the original image, remove original bindings jar.

**Comment by ravibeta** [comment]
Yes Sachin I did

File: docker/pravega/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+#FROM openjdk:8u212-jre-alpine3.9
+ARG DOCKER_IMAGE_REGISTRY
+ARG DOCKER_IMAGE_REPOSITORY
+ARG DOCKER_IMAGE_VERSION
+ARG PRAVEGA_IMAGE="${DOCKER_IMAGE_REGISTRY}/${DOCKER_IMAGE_REPOSITORY}:${DOCKER_IMAGE_VERSION}"
+FROM ${PRAVEGA_IMAGE}
+WORKDIR /opt/pravega
+ARG PRAVEGA_BINDINGS_VERSION
+ARG HDFS_JAR="hdfs/build/libs/pravega-hdfs-${PRAVEGA_BINDINGS_VERSION}.jar"
+ARG S3_JAR="extendeds3/build/libs/pravega-extendeds3-${PRAVEGA_BINDINGS_VERSION}.jar"
```
Comment (sachin-j-joshi) [inline docker/pravega/Dockerfile:20]: Where are we removing old bindings jar ?

**Review by sachin-j-joshi — CHANGES_REQUESTED** [review]

File: docker/pravega/Dockerfile (hunk 2 above)
Comment (ravibeta) [reply docker/pravega/Dockerfile:20]: Yes Sachin, it was there. Let me check. Thanks,

Comment (ravibeta) [reply docker/pravega/Dockerfile:20]: https://github.com/pravega/pravega-storage-adapters/pull/15/files#diff-ec695cab72c3c30c1f15b1bcd0dafbfcR24

File: docker/pravega/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+COPY ${HDFS_JAR} /opt/pravega/lib
+COPY ${S3_JAR} /opt/pravega/lib
+ARG ONLY
+RUN if [ -z "$ONLY" ]; then echo "keeping earlier bindings jar"; else rm -rf /opt/pravega/lib/pravega-bindings-*; fi
```
Comment (ravibeta) [inline docker/pravega/Dockerfile:24]: This is where we remove earlier bindings jar. Thanks

File: docker/pravega/Dockerfile (hunk 2 above)
Comment (ravibeta) [reply docker/pravega/Dockerfile:20]: Line 24

**Review by sachin-j-joshi — APPROVED** [review]
Looks okay

**Comment by ravibeta** [comment]
Yes Sachin Thank you very much. May I please ask you to merge this approved pr ? Thanks, Ravi

**Pull request merged**